site/content/blog/tilt-carvel-local-workflow.md

```diff
@@ -0,0 +1,205 @@
+---
+
+Software development often involves a cycle of making a code change, running unit tests, building an image then deploying a container
+to Docker or Kubernetes. [Tilt](https://docs.tilt.dev/index.html) is a tool that can help to automate the local workflow of
+```
+code -> build -> deploy -> test
+```
+
+In this article, we will take a tour of the capabilities of Tilt and demonstrate how it can be integrated with Carvel tools. 
+
+## What is Tilt?
+Tilt is a command line tool with a built-in server that continuously builds and deploys code by watching for changes.
+This allows developers to make code changes and run tests without having to manually run commands to deploy the change. 
+This is a great feature for reducing friction when running end-to-end tests during development. 
+Tilt supports deployment to local machine, docker, kubernetes and custom targets.  
+
+Tilt is configured using a [Starlark](https://github.com/bazelbuild/starlark) based language that will feel familiar to 
+[ytt](https://github.com/vmware-tanzu/carvel-ytt) users. By using a flexible configuration language, Tilt can easily
+be customized for individual needs and integrated with existing workflows.
+
+## Tools you will need to run the demo
+- [tilt cli](https://docs.tilt.dev/index.html)
+- [go](https://go.dev/doc/install) for building the demo
+- Kubernetes cluster such as [kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installing-with-a-package-manager)
+- [kapp](https://github.com/vmware-tanzu/carvel-kapp)
+- [ytt](https://github.com/vmware-tanzu/carvel-ytt)
+- [kbld](https://github.com/vmware-tanzu/carvel-kbld)
+
+## Checkout the demo project
+
+First, [checkout the project from GitHub](https://github.com/ojhughes/carvel-tilt-example-go)
+
+The repository contains a simple Go webserver and a set of manifests for deploying the application to Kubernetes
+
+```
+├── Dockerfile
+├── README.md
+├── Tiltfile
+├── build
+│   └── carvel-tilt
+├── carvel-tilt-example-go.iml
+├── cmd
+│   └── carvel-tilt
+│       └── main.go
+├── deployments
+│   ├── build.yaml
+│   ├── deployment.yaml
+│   ├── namespace.yaml
+│   ├── service.yaml
+│   └── values-schema.yml
+├── go.mod
+├── pkg
+│   └── api
+│       └── handler.go
+```
+
+## Render Kubernetes manifests with ytt
+The demo uses ytt to render the Kubernetes YAML, this allows variables such as HTTP port to be substituted at build time.
+The YAML can be rendered with the command 
+```shell
+ytt -f deployments
+```
+A number of default values are defined in the file `deployments/values-schema.yml`, which can be easily overridden.
+To change the HTTP port of the deployed webserver, use command;
+
+```shell
+ytt -f deployments --data-value-yaml port=8085
+```
+Note, `--data-value-yaml` is used instead of `--data-value` because port is an integer value
+
+## Building the image with kbld
+`kbld` is used to build the image, this is a very useful tool for local development as it will automatically tag
```

Comment:
```suggestion
[Kbld](https://carvel.dev/kbld/) is used to build the image, this is a very useful tool for local development as it will automatically tag
```